File: backend/app/Models/Game.php
<?php

namespace App\Models;

use App\Enums\GameStatus;
use Illuminate\Database\Eloquent\Model;
use Illuminate\Database\Eloquent\Relations\BelongsTo;
use Illuminate\Support\Carbon;

/**
 * @property int $id
 * @property int $home_team_id
 * @property int $away_team_id
 * @property int $round_number
 * @property int|null $home_score
 * @property int|null $away_score
 * @property GameStatus $status
 * @property Carbon|null $kickoff_at
 */
class Game extends Model
{
    protected $table = 'matches';

    protected $fillable = [
        'home_team_id',
        'away_team_id',
        'round_number',
        'home_score',
        'away_score',
        'status',
        'kickoff_at',
    ];

    protected function casts(): array
    {
        return [
            'home_team_id' => 'integer',
            'away_team_id' => 'integer',
            'round_number' => 'integer',
            'home_score' => 'integer',
            'away_score' => 'integer',
            'status' => GameStatus::class,
            'kickoff_at' => 'datetime',
        ];
    }

    /**
     * @return BelongsTo<Team, $this>
     */
    public function homeTeam(): BelongsTo
    {
        return $this->belongsTo(
            Team::class,
            'home_team_id'
        );
    }

    /**
     * @return BelongsTo<Team, $this>
     */
    public function awayTeam(): BelongsTo
    {
        return $this->belongsTo(
            Team::class,
            'away_team_id'
        );
    }
}

File: mobile/lib/features/tournament/domain/game.dart
enum GameStatus {
  scheduled,
  inPlay,
  finished;

  factory GameStatus.fromApi(String value) {
    return switch (value) {
      'scheduled' => GameStatus.scheduled,
      'in_play' => GameStatus.inPlay,
      'finished' => GameStatus.finished,
      _ => throw FormatException('Unknown game status: $value'),
    };
  }

  String get apiValue {
    return switch (this) {
      GameStatus.scheduled => 'scheduled',
      GameStatus.inPlay => 'in_play',
      GameStatus.finished => 'finished',
    };
  }
}

class Game {
  const Game({
    required this.id,
    required this.roundNumber,
    required this.homeTeamId,
    required this.awayTeamId,
    required this.homeScore,
    required this.awayScore,
    required this.status,
    required this.kickoffAt,
  });

  final int id;
  final int roundNumber;

  final int homeTeamId;
  final int awayTeamId;

  final int? homeScore;
  final int? awayScore;

  final GameStatus status;

  final DateTime? kickoffAt;

  factory Game.fromJson(Map<String, dynamic> json) {
    return Game(
      id: json['id'] as int,
      roundNumber: json['round_number'] as int,
      homeTeamId: json['home_team_id'] as int,
      awayTeamId: json['away_team_id'] as int,
      homeScore: json['home_score'] as int?,
      awayScore: json['away_score'] as int?,
      status: GameStatus.fromApi(json['status'] as String),
      kickoffAt: json['kickoff_at'] == null
          ? null
          : DateTime.parse(json['kickoff_at'] as String),
    );
  }
}
